### PR TITLE
Create realm role page

### PR DESCRIPTION
**Pull request by jenny-s51**
This is a branch from #11. Towards #94

### CODE REVIEW

File: src/realm-roles/messages.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
     "roles": {
       "createRole": "Create role",
+      "create" : "Create",
```
Comment (ssilvert) [inline src/realm-roles/messages.json:4]: Something this generic probably belongs in `common-messages.json`

File: src/realm-roles/add/NewRoleForm.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+//     const obj = value ? JSON.parse(value as string) : defaultRealm;
+//     Object.keys(obj).forEach((k) => {
+//       setValue(k, obj[k]);
+//     });
+//   };
+
+  const save = async (role: RoleRepresentation) => {
+    try {
+      await httpClient.doPost(`/auth/admin/realms/${realm}/roles`, role);
```
Comment (ssilvert) [inline src/realm-roles/add/NewRoleForm.tsx:47]: Looks like this didn't actually work.

Your payload is wrong.  You are sending `{"role", "newrole"}`.  Should be `{"name", "newrole"}`

Easy way to double-check is to look at what the old console does.

**Review by ssilvert — CHANGES_REQUESTED** [review]

**Comment by jenny-s51** [comment]
@edewit @ssilvert do you guys have any ideas as to why the HTTP post doesn't work with this route?

**Comment by edewit** [comment]
> @edewit @ssilvert do you guys have any ideas as to why the HTTP post doesn't work with this route?

@ssilvert suspected that the payload is wrong, is that not the case?

**Review by ssilvert — COMMENTED** [review]
@jenny-s51 The Create roll function is working now.  Just a few minor things to fix and we can merge.

File: src/realm-roles/add/NewRoleForm.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  const { realm } = useContext(RealmContext);
+
+  const { register, handleSubmit } = useForm<RoleRepresentation>();
+
+  const save = async (role: RoleRepresentation) => {
+    try {
+      await httpClient.doPost(`admin/realms/${realm}/roles`, role);
+      add(t("Role created"), AlertVariant.success);
+    } catch (error) {
+      add(`${t("Could not create role:")} '${error}'`, AlertVariant.danger);
```
Comment (ssilvert) [inline src/realm-roles/add/NewRoleForm.tsx:35]: Use camel Case

```suggestion
      add(`${t("couldNotCreateRole")} '${error}'`, AlertVariant.danger);
```

**Review by ssilvert — CHANGES_REQUESTED** [review]

File: src/realm-roles/add/NewRoleForm.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            <TextInput
+              isRequired
+              type="text"
+              id="kc-role-name"
+              name="name"
+              ref={register()}
+            />
+          </FormGroup>
+          <FormGroup label={t("description")} fieldId="kc-role-description">
+            <TextInput
```
Comment (dlabaj) [inline src/realm-roles/add/NewRoleForm.tsx:60]: Did you create a PR against pattern fly for the TextArea failure?

**Review by dlabaj — APPROVED** [review]

File: src/realm-roles/add/NewRoleForm.tsx (hunk 4 above)
Comment (jenny-s51) [reply src/realm-roles/add/NewRoleForm.tsx:60]: yes - https://github.com/patternfly/patternfly-react/issues/4894

**Review by edewit — APPROVED** [review]

**Review by dlabaj — APPROVED** [review]

**Review by ssilvert — APPROVED** [review]

**Pull request merged**